plugins/env/app/models/external_environment_variable_group.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+require 'aws-sdk-s3'
+
+class ExternalEnvironmentVariableGroup < ActiveRecord::Base
+  S3_URL_REGEX = /\Ahttps:\/\/([^.]+)\.s3\.amazonaws\.com\/([\w\W]+)\Z/i.freeze
+  S3_URL_FORMAT = "https://[bucket].s3.amazonaws.com/[key]?versionId=[version_id]"
+  HELP_TEXT = ENV.fetch(
+    "EXTERNAL_ENV_GROUP_HELP_TEXT",
+    "Use external service to manage environment variable groups"
+  ).html_safe
+  attr_accessor :key, :bucket, :version_id
+  audited
+  default_scope -> { order(:name) }
+
+  belongs_to :project, inverse_of: :external_environment_variable_groups
+  validates :name, :url, presence: true
+  validates :url, format: {
+    with: S3_URL_REGEX,
+    message: "Invalid format, valid url format is #{S3_URL_FORMAT}"
+  }
+  validate :validate_s3_url
+  after_find :resolve_s3_url
+
+  def external_service_read_with_failover
+    default_bucket = ENV.fetch 'EXTERNAL_ENV_GROUP_S3_BUCKET'
+    default_region = ENV.fetch 'EXTERNAL_ENV_GROUP_S3_REGION'
+    dr_bucket      = ENV['EXTERNAL_ENV_GROUP_S3_DR_BUCKET']
+    dr_region      = ENV['EXTERNAL_ENV_GROUP_S3_DR_REGION']
+    if default_bucket != bucket && dr_bucket != bucket
+      buckets = [default_bucket, dr_bucket].compact
+      raise "Invalid s3 bucket, acceptable buckets are #{buckets.join(',')}"
```

Comment:
if we know the supported bucket, then this should happen during validation or users should only set the path

Reply:
Yes, this is validated during create and update